chore: remove unused benchmarks

majorgreys opened the pull request:
With #6275, these benchmarks are no longer run in CI. This is because we have migrated to a separate CI pipeline for running benchmarks. The benchmark scenarios are now in the top-level `benchmarks`.

## Checklist

- [x] Change(s) are motivated and described in the PR description.
- [x] Testing strategy is described if automated tests are not included in the PR.
- [x] Risk is outlined (performance impact, potential for breakage, maintainability, etc).
- [x] Change is maintainable (easy to change, telemetry, documentation).
- [x] [Library release note guidelines](https://ddtrace.readthedocs.io/en/stable/releasenotes.html) are followed. If no release note is required, add label `changelog/no-changelog`.
- [x] Documentation is included (in-code, generated user docs, [public corp docs](https://github.com/DataDog/documentation/)).
- [x] Backport labels are set (if [applicable](https://ddtrace.readthedocs.io/en/latest/contributing.html#backporting))

## Reviewer Checklist

- [x] Title is accurate.
- [x] No unnecessary changes are introduced.
- [x] Description motivates each change.
- [x] Avoids breaking [API](https://ddtrace.readthedocs.io/en/stable/versioning.html#interfaces) changes unless absolutely necessary.
- [x] Testing strategy adequately addresses listed risk(s).
- [x] Change is maintainable (easy to change, telemetry, documentation).
- [x] Release note makes sense to a user of the library.
- [x] Reviewer has explicitly acknowledged and discussed the performance implications of this PR as reported in the benchmarks PR comment.
- [x] Backport labels are set in a manner that is consistent with the [release branch maintenance policy](https://ddtrace.readthedocs.io/en/latest/contributing.html#backporting)

mabdinur reviewed (commented):
Why aren't these benchmarks run in ci? Can we get any value from enabling these benchmarks or using the microbenchmark platform to run these scenarios

majorgreys commented:
> Why aren't these benchmarks run in ci? Can we get any value from enabling these benchmarks or using the microbenchmark platform to run these scenarios

I updated the PR description with the PR that dropped the job in CI and added a reference to the benchmark scenarios we run as a separate CI pipeline
